src/app/[locale]/dataset-list.tsx

```diff
@@ -0,0 +1,38 @@
+'use client';
+
+import {useQuery} from '@tanstack/react-query';
+import {SearchResult} from '@/lib/dataset-fetcher';
+import DatasetCard from '@/components/dataset-card';
+
+interface Props {
+  initialSearchResult: SearchResult;
+  locale: string;
+}
+
+export default function DatasetList({initialSearchResult, locale}: Props) {
+  const queryResponse: {data: SearchResult} = useQuery({
+    queryKey: ['Datasets'],
+    queryFn: async () => {
+      const response = await fetch('/api/datasets');
```

Comment:
Afterthought: what happens if `fetch()` returns an error (i.e. `response.ok` is false)?

Reply:
I will take this remark to the next ticket, where I implement the filters.

Reply:
I've put this on the todo list in #46